lib/fluent/system_config.rb

```diff
@@ -95,13 +97,13 @@ def self.overwrite_system_config(hash)
       end
     end
 
-    def initialize(conf=nil)
+    def initialize(conf=nil, strict_config_value=false)
```

Comment:
When I use keyword argument for it, some tests are broken like this:

```
==============================================================================================================================================
/home/aho/Projects/fluent/fluentd/test/config/test_system_config.rb:100:in `block in <class:TestSystemConfig>'
/home/aho/Projects/fluent/fluentd/test/config/test_system_config.rb:100:in `new'
/home/aho/Projects/fluent/fluentd/lib/fluent/system_config.rb:103:in `initialize'
/home/aho/Projects/fluent/fluentd/lib/fluent/system_config.rb:107:in `configure'
/home/aho/Projects/fluent/fluentd/lib/fluent/configurable.rb:90:in `configure'
Error: test: accepts parameters[workers](Fluent::Config::TestSystemConfig): NoMethodError: undefined method `<<' for nil:NilClass
```

`nil` seems be passed for the first argument of `initialize` even if a caller passed a valid config.

It is resolved when I add the argument `strict_config_value: flase (or true)` to the caller, or remove the default value `nil` from `conf`.
I'm not sure Ruby's spec for this issue.

Reply:
> It is resolved when I add the argument strict_config_value: flase (or true) to the caller, or remove the default value nil from conf.

This is what I intended. the following two syntaxes should be almost the same. 
Or am I get wrong something?

```
def initialize(conf=nil, strict_config_value=false)
 ...
end

```

```
def initialize(conf=nil, strict_config_value: false)
 ...
end
```

Reply:
Probably one of the cause is that `Fluent::Config::Element` inherits Hash class.
On calling `SystemConfig::new(conf)`, `conf`is evaluated as keyword arguments at first since its a Hash object (like `SystemConfig::new({a: 1, b: 2})`). The conf object seems be broken while processing arguments by Ruby (maybe).

To avoid such confusion, keyword arguments shouldn't be used in this case.

Reply:
> why don't you use keyword arguments since you already did it at SystemConfig#configure and SystemConfig.create?

I've replace with a normal argument for them to unify the signature and to avoid confusion.